### PR TITLE
Refactored preprocessor and added argument parser

### PR DESCRIPTION
**Pull request by gpalsingh**

### CODE REVIEW

**Comment by lamerman** [comment]
The line limit is 120 symbols, there is no need to make it 80 :)

**Comment by lamerman** [comment]
I would stop for now on what we agreed yesterday, else it will be infinite review. Other things could be made in separate PRs

**Comment by gpalsingh** [comment]
i didnt do anything myself. i just used autopep8 . ill find the option for setting character limit.

**Comment by lamerman** [comment]
Ok

**Comment by gpalsingh** [comment]
Why didn't you merge?  you want me to set line lengths before that?

File: shellpython/preprocessor.py [hunk 1]
```diff
@@ -76,9 +77,16 @@ def _get_username():
...
     :return: The name of current user
     """
-    return getpass.getuser()
+    try:
+        n = getpass.getuser()
+        return n
+    except:
+        # returns temporary name. eg: temp1232433
+        from time import time
+        return 'temp' + str(int(time))
```
Comment (lamerman) [inline shellpython/preprocessor.py]: You removed it as I remember, this time based generation of user name

**Comment by lamerman** [comment]
Yes, without it there are too many changes. Please return the line lengths first.

File: shellpython/preprocessor.py (hunk 1 above)
Comment (gpalsingh) [inline shellpython/preprocessor.py]: oops!  it must have come back when I used rebase.

**Comment by gpalsingh** [comment]
That should work

**Comment by lamerman** [comment]
It seems ok now, but has some conflicts.

**Comment by gpalsingh** [comment]
That's because i squashed the commits i made. Otherwise, the code is same.

**Pull request closed without merging**